components/sync/engine/brave_syncer_device_poll_unittest.cc

```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/sync/engine/brave_syncer_device_poll.h"
+
+#include <memory>
+
+#include "base/functional/callback.h"
+#include "base/test/scoped_mock_clock_override.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace syncer {
+
+namespace {
+
+void MarkInvoked(bool* invoked) {
+  ASSERT_TRUE(invoked);
+  *invoked = true;
+}
+
+}  // namespace
+
+TEST(BraveSyncerDevicePoll, ForcedPolling) {
+  base::ScopedMockClockOverride clock;
+  BraveSyncerDevicePoll brave_syncer_device_poll;
+  const auto kDelayBeforeForcedPoll =
+      BraveSyncerDevicePoll::GetDelayBeforeForcedPollForTest();
+
+  bool invoked = false;
+  brave_syncer_device_poll.CheckIntervalAndPoll(
+      base::BindOnce(MarkInvoked, &invoked));
```

Comment:
please use `MockCallback` instead.

```
{
  base::test::MockCallback<void()> mock_callback;
  EXPECT_CALL(mock_callback, Run).Times(0);
  brave_syncer_device_poll.CheckIntervalAndPoll(callback.Get());
  testing::Mock::VerifyAndClearExpectations(&mock_callback);
}
```

Reply:
thanks, test is much cleaner now :+1: 
fixed at https://github.com/brave/brave-core/pull/17077/commits/8fdb74c0f295ae4cd356bf0cba3a4967c208c969